docs/reference/mapping/runtime.asciidoc

```diff
@@ -803,6 +808,115 @@ address.
 // TESTRESPONSE[s/"_id" : "oWs5KXYB-XyJbifr9mrz"/"_id": $body.hits.hits.0._id/]
 // TESTRESPONSE[s/"day_of_week" : \[\n\s+"Sunday"\n\s\]/"day_of_week": $body.hits.hits.0.fields.day_of_week/]
 
+
+[[lookup-runtime-fields]]
+==== Retrieve fields from related indices
+
+The <<search-fields,`fields`>> parameter on the `_search` API can also be used to retrieve fields from
+the related indices via runtime fields with a type of `lookup`.
+
+[source,console]
+----
+POST ip_location/_doc?refresh
+{
+  "ip": "192.168.1.1",
+  "country": "Canada",
+  "city": "Montreal"
+}
+
+PUT logs/_doc/1?refresh
+{
+  "host": "192.168.1.1",
+  "message": "the first message"
+}
+
+PUT logs/_doc/2?refresh
+{
+  "host": "192.168.1.2",
+  "message": "the second message"
+}
+
+POST logs/_search
+{
+  "runtime_mappings": {
+    "location": {
+        "type": "lookup", <1>
+        "index": "ip_location", <2>
+        "query_type": "term", <3>
```

Comment:
Small comment, it could be nice to exclude "query_type" in this first version to keep things simple. I'm actually not sure how it will look when we extend to other queries, would the user want to specify the query type like this, or use a real query definition plus a search-template like syntax?

Also what do you think of adding some structure similar to this? It could clarify what is associated with the lookup vs. main index.
```
"path": "host",
"lookup": {
    "index": "ip_location",
    "path": "ip",
    "fields": ["country", "city"]
}
```

Reply:
I found your suggestion is more confusing. I've changed `index` to `lookup_index`.

> it could be nice to exclude "query_type" in this first version to keep things simple.

I can do it, but I think we will introduce the`geo_match` query type soon.